js/app/modules/controller/course.js

```diff
@@ -0,0 +1,61 @@
+// Copyright 2015 Endless Mobile, Inc.
+
+/* exported Course */
+
+const GObject = imports.gi.GObject;
+
+const CourseHistoryStore = imports.app.courseHistoryStore;
+const Engine = imports.search.engine;
+const Controller = imports.app.interfaces.controller;
+const HistoryStore = imports.app.historyStore;
+const Module = imports.app.interfaces.module;
+const Pages = imports.app.pages;
+const ReadingHistoryModel = imports.app.readingHistoryModel;
+const SetMap = imports.app.setMap;
+const QueryObject = imports.search.queryObject;
+
+/**
+ * Class: Course
+ * Controller that presents content as a course to be consumed.
+ *
+ */
+const Course = new Module.Class({
+    Name: 'Controller.Course',
+    Extends: GObject.Object,
+    Implements: [Controller.Controller],
+
+    _init: function (props={}) {
+        this.parent(props);
+
+        let history = new CourseHistoryStore.CourseHistoryStore();
+        HistoryStore.set_default(history);
+
+        this._window = this.create_submodule('window', {
+            application: this.application,
+            visible: false,
+        });
+
+        this.load_theme();
+    },
+
+    make_ready: function (cb=function () {}) {
+        // Load all sets, with which to populate the set map
+        // FIXME: deduplicate this with Selection.AllSets
+        Engine.get_default().get_objects_by_query(new QueryObject.QueryObject({
```

Comment:
Controller parent class? There is no controller parent class - it's just an interface, and we don't have an `interface_make_ready()` now AFAIK